workflows/scripts/convert.py

```diff
@@ -0,0 +1,61 @@
+"""
+script: covert.py
+
+converts sqlite (or other formats) into parquet files
+"""
+import pathlib
+from typing import Union, List
+
+import cytotable
+
+
+def sqlite_to_parquet(
+    input_file: Union[str, List[str]],
+    out_path: str,
+    target_ext: str,
+    convert_configs: str,
+):
+    # checking if user has parquet file
+    if target_ext == ".parquet" and pathlib.Path(input_file).suffix != ".sqlite":
+        raise ValueError(
+            "Converting to parquet files requires sqlite file."
+            f"File provided: {target_ext}"
+        )
+
+    # convert sqlite file into parquet
+    cytotable_config = convert_configs["params"]
+    cytotable.convert(
+        source_path=input_file,
+        dest_path=out_path,
+        dest_datatype=cytotable_config["dest_datatype"],
+        source_datatype=cytotable_config["source_datatype"],
+        concat=cytotable_config["concat"],
+        join=cytotable_config["join"],
+        infer_common_schema=cytotable_config["infer_common_schema"],
+        drop_null=cytotable_config["drop_null"],
+        preset=cytotable_config["preset"],
+        log_level=cytotable_config["log_level"],
+    )
+
+
+def main():
```

Comment:
i agree - unless it is snakemake specific, please try to reduce complexity - this will help SO much in the future (avoiding headaches with debugging, updating for new functionality, general gardening, etc.)

Reply:
I agree, this is unnecessarily complex for just a single function call.

I've been thinking about this quite some time. Since the inputs are ["magically" imported](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#python) into the script, I feel that the main script should handle everything. 

I think this specific design will only work if we have multiple functions. As of right now, only one function is being used. 

I'll make this as an issue. #41

Reply:
@axiomcura - Thanks for addressing this, for now it might be helpful to link to the `magically imported` input documentation you mentioned. This could help rationalize future development decisions if they are needed before #41 is resolved.